firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -12,15 +12,22 @@ import kotlinx.coroutines.flow.filter
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.withTimeout
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.nullable
+import kotlinx.serialization.builtins.serializer
 import kotlin.random.Random
 import kotlin.test.*
 
 expect val emulatorHost: String
 expect val context: Any
 expect fun runTest(test: suspend CoroutineScope.() -> Unit)
 
+/** @return a map extracted from the encoded data. */
 expect fun encodedAsMap(encoded: Any?): Map<String, Any?>
+/** @return paris as raw encoded data. */
+expect fun rawEncoded(vararg pairs: Pair<String, Any?>): Any
```

Comment:
This rawEncoded name is confusing, because in the end you encode a map, right? And varargs is to save some typing. What is do other way around:

```
expect fun Map<String, Any?>.encoded(): Any
```

Easy to type: mapOf().encoded() easy to read